lib/_stream_transform.js

```diff
@@ -115,9 +103,16 @@ function Transform(options) {
 
   Duplex.call(this, options);
 
-  this._transformState = new TransformState(this);
-
-  var stream = this;
+  this._transformState = {
+    afterTransform: (er, data) => {
+      return afterTransform(this, er, data);
+    },
```

Comment:
can we try to use `bind` here? It might be slightly faster as we don't keep the context alive.